SNOW-763357: Following Setup of tests/readme.md does not work

Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

   Python 3.8.16

2. What operating system and processor architecture are you using?

   macOS-10.16-x86_64-i386-64bit

3. What are the component versions in the environment (`pip freeze`)?

   alabaster==0.7.13
  asn1crypto==1.5.1
  attrs==22.2.0
  Babel==2.12.1
  cachetools==5.3.0
  certifi @ file:///private/var/folders/sy/f16zz6x50xz3113nwtb9bvq00000gp/T/abs_477u68wvzm/croot/certifi_1671487773341/work/certifi
  cffi==1.15.1
  chardet==5.1.0
  charset-normalizer==2.1.1
  cloudpickle==2.0.0
  colorama==0.4.6
  coverage==7.2.1
  cryptography==39.0.2
  distlib==0.3.6
  docutils==0.18.1
  exceptiongroup==1.1.1
  filelock==3.9.1
  idna==3.4
  imagesize==1.4.1
  importlib-metadata==6.0.0
  iniconfig==2.0.0
  Jinja2==3.1.2
  MarkupSafe==2.1.2
  numpy==1.24.2
  oscrypto==1.3.0
  packaging==23.0
  pandas==1.5.3
  platformdirs==3.1.1
  pluggy==1.0.0
  pyarrow==10.0.1
  pycparser==2.21
  pycryptodomex==3.17
  Pygments==2.14.0
  PyJWT==2.6.0
  pyOpenSSL==23.0.0
  pyproject_api==1.5.1
  pytest==7.2.2
  pytest-cov==4.0.0
  python-dateutil==2.8.2
  pytz==2022.7.1
  requests==2.28.2
  six==1.16.0
  snowballstemmer==2.2.0
  snowflake-connector-python==3.0.1
  -e git+https://github.com/suenalaba/snowpark-python.git@3533cb9a00d2958b6ec2b41de67f30b6b2f79eb2#egg=snowflake_snowpark_python
  Sphinx==5.0.2
  sphinxcontrib-applehelp==1.0.4
  sphinxcontrib-devhelp==1.0.2
  sphinxcontrib-htmlhelp==2.0.1
  sphinxcontrib-jsmath==1.0.1
  sphinxcontrib-qthelp==1.0.3
  sphinxcontrib-serializinghtml==1.1.5
  tomli==2.0.1
  tox==4.4.7
  typing_extensions==4.5.0
  urllib3==1.26.15
  virtualenv==20.21.0
  zipp==3.15.0

4. What did you do?
   In the project directory, running `pytest tests/integ/test_dataframe.py` does not work
   but running `pytest tests/unit/test_dataframe.py` works.

5. What did you expect to see?

   Integration Test Results with passed and failed numbers.

6. Can you set logging to DEBUG and collect the logs?

   The documentation described in: https://github.com/snowflakedb/snowpark-python/blob/main/tests/README.md
   only allows unit tests to work. 
   In DB parameters, 
            host=ret["host"],
            port=ret["port"],
            protocol=ret["protocol"],
   requires the host, port and protocol to be set for the integration tests to run.

<!--
If you need urgent assistance reach out to support for escalated issue processing https://community.snowflake.com/s/article/How-To-Submit-a-Support-Case-in-Snowflake-Lodge
-->

Hi @sfc-gh-jdu thanks for getting back to me. I think my information should be all correct, as I am able to run Jupyter Notebook scripts demo and unit tests for example using: `pytest tests/unit/test_dataframe.py`. 

This error only appears when i try to run integ tests using: `pytest tests/integ/test_dataframe.py`.

The following error log is obtained for all the integration tests:
```
    @pytest.fixture(scope="session")
    def connection(db_parameters):
        ret = db_parameters
        with snowflake.connector.connect(
            user=ret["user"],
            password=ret["password"],
>           host=ret["host"],
            port=ret["port"],
            database=ret["database"],
            account=ret["account"],
            protocol=ret["protocol"],
            role=ret.get("role"),
        ) as con:
E       KeyError: 'host'

tests/integ/conftest.py:60: KeyError
```
All the integration tests fail due to the key error.

Apparently, the port, protocol and host is needed?

--- a/tests/integ/conftest.py
+++ b/tests/integ/conftest.py
@@ -55,13 +55,13 @@ def resources_path() -> str:
 def connection(db_parameters):
     ret = db_parameters
     with snowflake.connector.connect(
-        user=ret["user"],
-        password=ret["password"],
-        host=ret["host"],
-        port=ret["port"],
-        database=ret["database"],
-        account=ret["account"],
-        protocol=ret["protocol"],
+        user=ret.get("user"),
+        password=ret.get("password"),
+        host=ret.get("host"),
+        port=ret.get("port"),
+        database=ret.get("database"),
+        account=ret.get("account"),
+        protocol=ret.get("protocol"),
         role=ret.get("role"),
     ) as con:
         yield con

